x-pack/dockerlogbeat/pipelinemanager/pipelineManager.go

```diff
@@ -161,6 +247,35 @@ func (pm *PipelineManager) registerClient(cl *ClientLogger, hash uint64, clientF
 	pm.pipelines[hash].refCount++
 }
 
+// registerLogger registers a local logger used for reading back logs
+func (pm *PipelineManager) registerLogger(log logger.Logger, info logger.Info) {
+	pm.mu.Lock()
+	defer pm.mu.Unlock()
+	pm.clientLogger[info.ContainerID] = log
+}
+
+// removeLogger removes a logging instace
+func (pm *PipelineManager) removeLogger(info logger.Info) {
+	pm.mu.Lock()
+	defer pm.mu.Unlock()
+	logger, exists := pm.clientLogger[info.ContainerID]
+	if !exists {
+		return
+	}
+	logger.Close()
+	delete(pm.clientLogger, info.ContainerID)
+	if os.Getenv("DESTROY_LOGS_ON_STOP") == "true" {
```

Comment:
I'd prefer to read this variable once at init time. Being an env variable it can't be changed for the running process from the outside anyways.
Also check if the env variable is correct when parsing and log an error when not. I guess acceptable values are `"true", "false", ""`.